src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -44,7 +50,18 @@ public UserRegistrationResponseDto register(UserRegistrationRequestDto request)
         roles.add(defaultRole);
         user.setRoles(roles);
         User savedUser = userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();
```

Comment:
I would extract this logic to service related to shopping cart logic and call service here